File: app/src/main/java/com/example/dayo/data/database/UserDao.java
package com.example.dayo.data.database;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface UserDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertUser(User user);

    @Query("SELECT * FROM users WHERE email = :email AND password = :password")
    User getUserByEmailAndPassword(String email, String password);

    @Query("SELECT * FROM users WHERE id = :userId")
    User getUserById(int userId);

    @Query("SELECT * FROM users WHERE email = :email OR name = :username")
    User getUserByEmailOrUsername(String email, String username);
}

File: app/src/main/java/com/example/dayo/data/database/UserDao.kt
package com.example.dayo.data.database

import androidx.room.*

@Dao
interface UserDao {
    @Insert
    suspend fun insertUser(user: User)

    @Query("SELECT * FROM users WHERE id = :userId")
    suspend fun getUserById(userId: Int): User?

    @Query("SELECT * FROM users")
    suspend fun getAllUsers(): List<User>

    @Query("SELECT * FROM users WHERE :category IN (preferences)")
    suspend fun getUsersByCategory(category: Category): List<User>

    @Delete
    suspend fun deleteUser(user: User)
}